algorithms/sort/quick_sort/src/binary_quick_sort.rs

```diff
@@ -0,0 +1,185 @@
+#![allow(unused)]
+
+/// partition slice with pivot at end
+fn partition<T: Ord>(slice: &mut [T]) -> usize {
+    let len = slice.len();
+    let pivot = len - 1;
+    let mut cur_left_pos = 0usize;
+
+    for i in 0..len {
+        if slice[i] <= slice[pivot] {
+            slice.swap(cur_left_pos, i);
+            cur_left_pos += 1;
+        }
+    }
+    cur_left_pos - 1
+}
```

Comment:
```suggestion
fn partition<T: Ord>(slice: &mut [T]) -> usize {
    partition_by(slice, &mut Ord::cmp)
}
```
동일 로직이기 때문에 함수 전달로 구현해도 충분하고 유지보수에 용이할 것 같습니다.